Mathlib/MeasureTheory/Constructions/Projective.lean

```diff
@@ -0,0 +1,146 @@
+/-
+Copyright (c) 2023 Rémy Degenne. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Peter Pfaffelhuber
+-/
+import Mathlib.MeasureTheory.Constructions.Cylinders
+
+/-!
+# Projective measure families and projective limits
+
+A family of measures indexed by finite sets of `ι` is projective if, for finite sets `J ⊆ I`,
+the projection from `∀ i : I, α i` to `∀ i : J, α i` maps `P I` to `P J`.
+A measure `μ` is the projective limit of such a family of measures if for all `I : Finset ι`,
+the projection from `∀ i, α i` to `∀ i : I, α i` maps `μ` to `P I`.
+
+## Main definitions
+
+* `MeasureTheory.IsProjectiveMeasureFamily`: `P : ∀ J : Finset ι, Measure (∀ j : J, α j)` is
+  projective if the projection from `∀ i : I, α i` to `∀ i : J, α i` maps `P I` to `P J`
+  for all `J ⊆ I`.
+* `MeasureTheory.IsProjectiveLimit`: `μ` is the projective limit of the measure family `P` if for
+  all `I : Finset ι`, the map of `μ` by the projection to `I` is `P I`.
+
+## Main statements
+
+* `MeasureTheory.IsProjectiveLimit.unique`: the projective limit of a family of finite measures
+  is unique.
+
+-/
+
+open Set
+
+namespace MeasureTheory
+
+variable {ι : Type*} {α : ι → Type*} [∀ i, MeasurableSpace (α i)]
+  {P : ∀ J : Finset ι, Measure (∀ j : J, α j)}
+
+/-- A family of measures indexed by finite sets of `ι` is projective if, for finite sets `J ⊆ I`,
+the projection from `∀ i : I, α i` to `∀ i : J, α i` maps `P I` to `P J`. -/
+def IsProjectiveMeasureFamily [∀ i, MeasurableSpace (α i)]
+    (P : ∀ J : Finset ι, Measure (∀ j : J, α j)) : Prop :=
+  ∀ (I J : Finset ι) (hJI : J ⊆ I),
+    P J = (P I).map (fun (x : ∀ i : I, α i) (j : J) ↦ x ⟨j, hJI j.2⟩)
+
+namespace IsProjectiveMeasureFamily
+
+variable {I J : Finset ι}
+
+lemma congr_cylinder_of_subset [Nonempty (∀ i, α i)] (hP : IsProjectiveMeasureFamily P)
```

Comment:
I'm not sure why you would need a nonemptyness assumption here: if the product space is empty, then all the measures are zero (by `measure_univ_eq` below), so the conclusion is still true.